lib/specifier.js

```diff
@@ -120,6 +120,12 @@ function pick(versions, specifier, options) {
 }
 
 function satisfies(version, specifier, options = {}) {
+  if (specifier.startsWith("~=")) {
+    if (Operator.eq(version, specifier.replace(/^~=\s*/, ""))) {
+      return true;
+    }
+  }
+
```

Comment:
I don't think we have to go deeper and alter the parser behavior: side-effects can be bad